unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Material/AssetPromise_Material.cs

```diff
@@ -1,17 +1,201 @@
+            
+            CoroutineStarter.Stop(loadCoroutine);
+            loadCoroutine = null;
+        }
+
+        protected override void OnLoad(Action OnSuccess, Action<Exception> OnFail)
+        {
+            CoroutineStarter.Stop(loadCoroutine);
+            loadCoroutine = CoroutineStarter.Start(CreateMaterial(model, OnSuccess, OnFail));
+        }
+        
+        public override object GetId() { return model; }
+
+        private IEnumerator CreateMaterial(MaterialModel model, Action OnSuccess, Action<Exception> OnFail)
+        {
+            Material material = new Material(Utils.EnsureResourcesMaterial(MATERIAL_RESOURCES_PATH + PBR_MATERIAL_NAME));
+#if UNITY_EDITOR
+            material.name = "PBRMaterial_" + model.GetHashCode();
+#endif
+            material.SetColor(ShaderUtils.BaseColor, model.albedoColor);
+
+            if (model.emissiveColor != Color.clear && model.emissiveColor != Color.black)
+            {
+                material.EnableKeyword("_EMISSION");
+            }
+
+            // METALLIC/SPECULAR CONFIGURATIONS
+            material.SetColor(ShaderUtils.EmissionColor, model.emissiveColor * model.emissiveIntensity);
+            material.SetColor(ShaderUtils.SpecColor, model.reflectivityColor);
+
+            material.SetFloat(ShaderUtils.Metallic, model.metallic);
+            material.SetFloat(ShaderUtils.Smoothness, 1 - model.roughness);
+            material.SetFloat(ShaderUtils.EnvironmentReflections, model.microSurface);
+            material.SetFloat(ShaderUtils.SpecularHighlights, model.specularIntensity * model.directIntensity);
+
+
+            if (model.emissiveTexture != null)
+                emissionAssetPromiseDclTexture = AssignTextureToMaterial(material, ShaderUtils.EmissionMap, model.emissiveTexture, OnFail);
+
+            SetupTransparencyMode(material, model);
+
+            if (model.alphaTexture != null)
+                alphaAssetPromiseDclTexture = AssignTextureToMaterial(material, ShaderUtils.AlphaTexture, model.alphaTexture, OnFail);
+
+            if (model.albedoTexture != null)
+                baseAssetPromiseDclTexture = AssignTextureToMaterial(material, ShaderUtils.BaseMap, model.albedoTexture, OnFail);
+
+            if (model.bumpTexture != null)
+                bumpPromiseDclTexture = AssignTextureToMaterial(material, ShaderUtils.BumpMap, model.bumpTexture, OnFail);
+
+            // Checked two times so they can be loaded at the same time
+            if (model.alphaTexture != null)
+                yield return  alphaAssetPromiseDclTexture;
+            if (model.albedoTexture != null)
+                yield return  baseAssetPromiseDclTexture;
+            if (model.bumpTexture != null)
+                yield return  bumpPromiseDclTexture;
+            if (model.emissiveTexture != null)
+                yield return  emissionAssetPromiseDclTexture;
+
+            asset.material = material;
```

Comment:
After the material is constructed, is *very important* to use `SRPBatchingHelper.OptimizeMaterial` on it. If that is not used then the SRP batching will not be effective, and this can have a huge impact in draw calls.

Reply:
Added!